init.lua

```diff
@@ -221,107 +222,109 @@ minetest.register_node("anvil:anvil", {
 		local wielded = puncher:get_wielded_item()
 		local meta = minetest.get_meta(pos)
 		local inv  = meta:get_inventory()
-		
-		if wielded:get_count() == 0 then
-			if not inv:is_empty("input") then
-				local return_stack = inv:get_stack("input", 1)
-				inv:set_stack("input", 1, nil)
-				local wield_index = puncher:get_wield_index()
-				puncher:get_inventory():set_stack("main", wield_index, return_stack)
-				remove_item(pos, node)
-			end		
-		end
-		
-		-- only punching with the hammer is supposed to work
-		if wielded:get_name() ~= 'anvil:hammer' then
-			return
-		end
-		
-		local input = inv:get_stack('input',1)
-	
-		-- only tools can be repaired
-		if( not( input ) 
-		or input:is_empty()
-				or input:get_name() == "technic:water_can" 
-				or input:get_name() == "technic:lava_can" ) then
-			return
-		end
-	
-		-- 65535 is max damage
-		local damage_state = 40-math.floor(input:get_wear()/1638)
-	
-		local tool_name = input:get_name()
+		if meta:get_string("owner") == puncher:get_player_name() then
```

Comment:
```Lua
if meta:get_string("owner") ~= puncher:get_player_name() then
	return
end
```
Less indents, same effect, better readability.

Reply:
thanks, I think I know now what you mean